test/support/sidekiq_test_helpers.rb

```diff
@@ -5,7 +5,8 @@ module SidekiqTestHelpers
   def with_real_sidekiq
     Sidekiq::Testing.disable! do
       Sidekiq.configure_client do |config|
-        config.redis = { namespace: "whitehall-test" }
+        parallel_test_runner_number = ActiveRecord::Base.connection.current_database.split("-").last.to_i
+        config.redis = { db: parallel_test_runner_number }
```

Comment:
It might be worth adding a comment to explain the slightly odd way we're getting the worker number.

```suggestion
        # The Rails built-in test parallelization doesn't make it easy to find the worker number.
        # So here we're using the number suffixed to the Postgres database name, because that is unique for each worker.
        parallel_test_runner_number = ActiveRecord::Base.connection.current_database.split("-").last.to_i
        config.redis = { db: parallel_test_runner_number }
```